Blockly Factory: Global Self Bug Fix

quachtina96 opened the pull request:
The Factory Utils function warnIfUnsavedChanges contained code copied from AppController that referenced self.blockLibraryController. The code worked properly despite self not being defined in the function because of an accidental global self defined. 

Changes made:
- made the global self local to the App Controller init function
- make the block library controller a parameter in the Factory Utils function.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly/616)
<!-- Reviewable:end -->

quachtina96 commented:
@evd2014 @picklesrus @vicng

picklesrus commented on demos/blocklyfactory/factory_utils.js:
```diff
@@ -988,11 +988,13 @@ FactoryUtils.savedBlockChanges = function(blockLibraryController) {
  * @return {boolean} Whether or not to proceed.
  */
-FactoryUtils.warnIfUnsavedChanges = function() {
+FactoryUtils.warnIfUnsavedChanges = function(blockLibraryController) {
```
Hmm, Does this function belong here?   Maybe it goes in the blockLibraryController instead?

quachtina96 commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[demos/blocklyfactory/factory_utils.js, line 995 \[r1\]](https://reviewable.io:443/reviews/google/blockly/616#-KQYJXGfN0DCIcd-zwOW-r1-995:-KQYK9MQNf6Z6KU1H-v_:bpnl2u) ([raw file](https://github.com/google/blockly/blob/4e38029ad49178a3477630999aaf9b4e9873175b/demos/blocklyfactory/factory_utils.js#L995)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> Hmm, Does this function belong here?   Maybe it goes in the blockLibraryController instead?  
</details>
Good call. The line was blurry here because warnIfUnsavedChanges reaches into the Block Factory. To improve this and avoid repeated code,  I moved the call to isStarterBlock to savedBlockChanges as you suggested before and moved this warnIfUnsavedChanges to the block library controller. Thank you!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/616)*
<!-- Sent from Reviewable.io -->

picklesrus commented:
LGTM. Merging!
